azure-pipelines.yml

```diff
@@ -38,35 +28,41 @@ jobs:
   - pwsh: build/Update-Headers.ps1 -Verify
     displayName: Verify headers
 
+  # Restore solution
+  - script: dotnet restore -p:Configuration=$(BuildConfiguration)
+    displayName: Restore solution
+
   # Build solution
-  - script: dotnet build -c Release
+  - script: dotnet build --no-restore -c $(BuildConfiguration)
```

Comment:
`build` implicitly restores. Does the static graph restore need an explicit restore?

Reply:
Instead of cramming everything into a single step, I'm separating it, so that it'll be easy to look into the logs in CI.